src/v2/controllers/vdeployment_controller/model/reconciler.rs

```diff
@@ -0,0 +1,278 @@
+    let namespace = vd.metadata.namespace.unwrap();
+    match &state.reconcile_step {
+        VDeploymentReconcileStepView::Init => {
+            let req = APIRequest::ListRequest(ListRequest {
+                kind: VReplicaSetView::kind(),
+                namespace: namespace,
+            });
+            let state_prime = VDeploymentReconcileState {
+                reconcile_step: VDeploymentReconcileStepView::AfterGetReplicaSets,
+                new_vrs: None,
+                old_vrs_list: Seq::<VReplicaSetView>::empty(),
+            };
+            (state_prime, Some(RequestView::KRequest(req)))
+        },
+        VDeploymentReconcileStepView::AfterGetReplicaSets => {
+            if !(resp_o.is_Some() && resp_o.get_Some_0().is_KResponse()
+            && resp_o.get_Some_0().get_KResponse_0().is_ListResponse()
+            && resp_o.get_Some_0().get_KResponse_0().get_ListResponse_0().res.is_ok()) {
+                (error_state(state), None)
+            } else {
+                let objs = resp_o.unwrap().get_KResponse_0().get_ListResponse_0().res.unwrap();
+                let vrs_list_or_none = objects_to_vrs_list(objs);
+                if vrs_list_or_none.is_none() { // error in unmarshalling
+                    (error_state(state), None)
+                } else {
+                    let (new_vrs_list, old_vrs_list) = filter_old_and_new_vrs(filter_vrs_list(vrs_list_or_none.unwrap(), vd), vd);
+                    (
+                        VDeploymentReconcileState {
+                            reconcile_step: VDeploymentReconcileStepView::RollReplicas,
+                            new_vrs: if new_vrs_list.len() > 0 { Some(new_vrs_list[0]) } else { None },
+                            old_vrs_list: old_vrs_list,
+                        },
+                        None,
+                    )
+                }
+            }
+        },
+        VDeploymentReconcileStepView::RollReplicas => {
+            if state.new_vrs.is_None() {
+                // create the new vrs
+                let new_vrs = make_replica_set(vd);
+                let req = APIRequest::CreateRequest(CreateRequest {
+                    namespace: namespace,
+                    obj: new_vrs.marshal(),
+                });
+                let state_prime = VDeploymentReconcileState {
+                    reconcile_step: VDeploymentReconcileStepView::RollReplicas,
+                    new_vrs: Some(new_vrs),
+                    ..state
+                };
+                (state_prime, Some(RequestView::KRequest(req)))
+            } else {
+                // update existing new vrs
+                if !state.new_vrs.unwrap().well_formed() {
+                    (error_state(state), None)
+                } else {
+                    let new_vrs = state.new_vrs.get_Some_0(); // k8s deterministically chooses the "oldest" vrs, we just use the first one
```

Comment:
Can you put a `TODO` here for `k8s deterministically chooses the "oldest" vrs, we just use the first one`?